eng/Testing.targets

```diff
@@ -50,7 +50,8 @@
     <SkipTests>true</SkipTests>
 
     <!-- Only run tests if the build is running on GitHub Actions  -->
-    <SkipTests Condition=" '$(IsGitHubActionsRunner)' == 'true' and '$(RunOnGithubActions)' == 'true' ">false</SkipTests>
+    <SkipTests Condition=" '$(IsGitHubActionsRunner)' == 'true' and '$(BuildOs)' == 'windows' and '$(RunOnGithubActionsWindows)' == 'true' ">false</SkipTests>
+    <SkipTests Condition=" '$(IsGitHubActionsRunner)' == 'true' and '$(BuildOs)' != 'windows' and '$(RunOnGithubActionsLinux)' == 'true' ">false</SkipTests>
 
     <!-- Only run tests if the build is running on Helix infra  -->
     <SkipTests Condition=" '$(IsAzdoHelixRunner)' == 'true' and '$(RunOnAzdoHelix)' == 'true' ">false</SkipTests>
```

Comment:
The issue affects only the GHA, I don't want to upset and break the AzDO builds (I tried changing the other properties, and that led to failed build in AzDO). 
We can look at the AzDO separation later.